contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c

```diff
@@ -401,9 +404,86 @@ PrintTupPrepareInfo(DR_printtup *myState, TupleDesc typeinfo, int numAttrs)
 	}
 }
 
+static int32
+resolve_numeric_typmod_from_append_or_mergeappend(Plan *plan, AttrNumber attno)
+{
+	ListCell	*lc;
+	uint8_t		max_precision = 0,
+				max_scale = 0,
+				precision = 0,
+				scale = 0,
+				integralDigitCount = 0;
+	int32		typmod = -1,
+				result_typmod = -1;
+	List		*planlist = NIL;
+	if (IsA(plan, Append))
+	{
+		planlist = ((Append *) plan)->appendplans;
+	}
+	else if(IsA(plan, MergeAppend))
+	{
+		planlist = ((MergeAppend *) plan)->mergeplans; 
+	}
+
+	Assert(planlist != NIL);
+	foreach(lc, ((Append *) plan)->appendplans)
+	{
+		Plan *outerplan = (Plan *) lfirst(lc);
+		TargetEntry *tle = get_tle_by_resno(outerplan->targetlist, attno);
+		if (IsA(tle->expr, Var))
+		{
+			Var *var = (Var *)tle->expr;
+			if (var->varno == OUTER_VAR)
+			{
+				typmod = resolve_numeric_typmod_outer_var(outerplan, var->varattno);
+			}
+		}
+		typmod = resolve_numeric_typmod_from_exp(outerplan, (Node *)tle->expr);
+		if (typmod == -1)
+			continue;
+		scale = (typmod - VARHDRSZ) & 0xffff;
+		precision = ((typmod - VARHDRSZ) >> 16) & 0xffff;
+		integralDigitCount = Max(precision - scale, max_precision - max_scale);
+		max_scale = Max(max_scale, scale);
+		max_precision = integralDigitCount + max_scale;
```

Comment:
what is the logic behind mixing precision and scale across the plans?

Reply:
This logic will ensure that it can accommodate max scale and max precision across the plans. For example, numeric(6,4) and numeric(6,3), final result would be numeric(7,4)

Reply:
does SQL server have same behavior? What if result is beyond max supported by SQL server?